src/realm/collection_parent.hpp

```diff
@@ -105,6 +107,7 @@ class CollectionParent : public std::enable_shared_from_this<CollectionParent> {
     static constexpr size_t s_max_level = 100;
 #endif
     size_t m_level = 0;
+    mutable size_t m_parent_version = 0;
```

Comment:
what is this used for? (and what issue are we fixing by introducing it?)

Reply:
Kind of hard to explain. It was adding the line with the comment "// Idempotent, but updates dict accessor" that triggered an error. That call will update the `dict` accessor. Before this fix, the `dict2` would ask its parent (`dict`) if it needs an update, and if it does not, it will not refresh either - which it should. So we need some kind of counter that the parent will increment on any refresh that can be compared with a corresponding counter in the child.